### PR TITLE
Fix stop cluster informer when a cluster is removed

### PR DESCRIPTION
**Pull request by pigletfly**
Signed-off-by: pigletfly <wangbing.adam@gmail.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None

```

### CODE REVIEW

File: pkg/controllers/status/workstatus_controller.go [hunk 1]
```diff
@@ -432,8 +432,6 @@ func (c *WorkStatusController) getGVRsFromWork(work *workv1alpha1.Work) (map[sch
 // getSingleClusterManager gets singleClusterInformerManager with clusterName.
 // If manager is not exist, create it, otherwise gets it from map.
 func (c *WorkStatusController) getSingleClusterManager(cluster *clusterv1alpha1.Cluster) (informermanager.SingleClusterInformerManager, error) {
-	// TODO(chenxianpao): If cluster A is removed, then a new cluster that name also is A joins karmada,
-	//  the cache in informer manager should be updated.
```
Comment (RainbowMango) [inline pkg/controllers/status/workstatus_controller.go:436]: Seems the informer for `unregistered-cluster` will be cleaned up at [cluster status controller](https://github.com/karmada-io/karmada/blob/27f5bc55c1a150b929ac22337772d2ba7f78b9b0/pkg/controllers/status/cluster_status_controller.go#L95).

This is outdated comments I guess.

Comment (pigletfly) [reply pkg/controllers/status/workstatus_controller.go:436]: yes

Comment (RainbowMango) [reply pkg/controllers/status/workstatus_controller.go:436]: Maybe I don't think so. Why do you think that? 
They are sharing the same `informer manager`, stop once would be enough.

**Pull request closed without merging**